src/mlpack/methods/neighbor_search/ns_model.hpp

```diff
@@ -13,12 +13,24 @@
 #include <mlpack/core/tree/binary_space_tree.hpp>
 #include <mlpack/core/tree/cover_tree.hpp>
 #include <mlpack/core/tree/rectangle_tree.hpp>
-
+#include <boost/variant.hpp>
```

Comment:
You're right, it is header only.  I am not sure if it is possible to have a Boost installation where some of the header-only libraries (but not all) are available; I haven't played with b2 (Boost's build system) enough recently to be sure.  If it's possible to have a Boost installation with some of the header-only libraries omitted, then we should include `variant` in there; otherwise, I don't think it's necessary.  But I'm not sure whether or not that's possible...